### PR TITLE
fix #825

### LINKED ISSUE #825: [1.2.0] panic: runtime error: slice bounds out of range [13:12]
```
panic: runtime error: slice bounds out of range [13:12]

goroutine 7 [running]:
github.com/buger/goreplay/proto.HasResponseTitle(0xc00027ec66, 0xb7, 0xb7, 0xc002610200)
	/go/src/github.com/buger/goreplay/proto/proto.go:357 +0x242
main.startHint(0xc002c12370, 0xc0003a0630)
	/go/src/github.com/buger/goreplay/input_raw.go:203 +0x7c
github.com/buger/goreplay/tcp.(*MessagePool).Handler(0xc0003a2a40, 0x1124fc0, 0xc003d5da20)
	/go/src/github.com/buger/goreplay/tcp/tcp_message.go:176 +0x9fa
github.com/buger/goreplay/capture.(*Listener).Listen(0xc00034f4a0, 0x1118860, 0xc0003a2a80, 0xc000380400, 0x0, 0x0)
	/go/src/github.com/buger/goreplay/capture/capture.go:163 +0x163
github.com/buger/goreplay/capture.(*Listener).ListenBackground.func1(0xc00011efc0, 0xc00034f4a0, 0x1118860, 0xc0003a2a80, 0xc000380400)
	/go/src/github.com/buger/goreplay/capture/capture.go:173 +0x7b
created by github.com/buger/goreplay/capture.(*Listener).ListenBackground
	/go/src/github.com/buger/goreplay/capture/capture.go:171 +0x84
```

### PR DESCRIPTION
**Pull request by urbanishimwe**
this fix some edge cases that may arise in parsing the HTTP response title

### CODE REVIEW

File: proto/proto_test.go [hunk 1]
```diff
@@ -330,7 +330,8 @@ func TestHasResponseTitle(t *testing.T) {
 		"HTTP/1.1 100 Continue\r\n": true,
 		"HTTP/1.1  \r\n":            false,
 		"HTTP/4.0 100Continue\r\n":  false,
-		"HTTP/4.0 100 Continue\r\n": false,
+		"HTTP/1.0 10r Continue\r\n": false,
+		"HTTP/1.1 200\r\n":          false,
```
Comment (cofyc) [inline proto/proto_test.go:334]: can you add this test case `HTTP/1.1 200\r\nServer: Tengine\r\nContent-Length: 0\r\nConnection: close\r\n`?

the current implementation does not ok with the input `HTTP/1.1 200\r\n` but will panic with `HTTP/1.1 200\r\nServer: Tengine\r\nContent-Length: 0\r\nConnection: close\r\n`

Comment (urbanishimwe) [reply proto/proto_test.go:334]: I added the test case and it has passed!

**Review by cofyc — APPROVED** [review]
LGTM

**Comment by cofyc** [comment]
but, gor 1.2.0 seems to have a bug in parsing network packets.. after this patch, requests received by our middleware are still incomplete.

I've increased `--input-raw-buffer-size` and `--output-http-response-buffer`. Right now, I can only use `gor` 1.1.0 in our environment.

**Pull request merged**